src/Workspaces/Core/Portable/FindSymbols/SymbolAndProjectId.cs

```diff
@@ -0,0 +1,175 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+
+namespace Microsoft.CodeAnalysis.FindSymbols
+{
+    /// <summary>
+    /// Represents a symbol and the project it was acquired from.
+    /// It should always be the case that if you have the original solution
+    /// that this symbol came from, that you'll be able to find this symbol
+    /// in the compilation for the specified project.
+    /// </summary>
+    internal struct SymbolAndProjectId
+    {
+        public readonly ISymbol Symbol;
+        public readonly ProjectId ProjectId;
+
+        public SymbolAndProjectId(ISymbol symbol, ProjectId projectId)
+        {
+            Symbol = symbol;
+            ProjectId = projectId;
+        }
+
+        public override bool Equals(object obj) => Equals((SymbolAndProjectId)obj);
+
+        public bool Equals(SymbolAndProjectId other)
+        {
+            return Equals(this.Symbol, other.Symbol);
```

Comment:
Good question (and i'll doc).  Basically, from teh perspective of all of our features, we don't care where teh symbol came from.  i.e. i don't want to find the same 'System.Foo' type in 10 different projects, and then find references to all of them.  Instead, i think of these as the same.  I only need the project-id to allow me to serialize/deserialize, and i only need one of those project IDs to do so.